cha/design/sync.tex

```diff
@@ -0,0 +1,97 @@
+        %NTP(external sync/offset calculations)
+\section{Methods}
+The following is a list of the possible solutions we have considered:
+\begin{itemize}
+    \item GPS
+    \item Audio
+    \item Light cues.
+    \item \ac{NTP}
+    \item \ac{PTP}
+\end{itemize}
+An important thing to realize is also that if we were to synchronize the clocks, as in change the actual clock on the phones, they are required to be rooted for an application to do so.
+While we placed \ac{UX} as a non-requirement, we can not expect users to root their device as, depending on country, it may violate copyright, warranty, and/or terms on service on the phone.
+As such we will strictly speaking not be synchronizing the clocks, rather we will keep track of each devices offset to the source we want to synchronize with.
+
+\subsubsection{GPS}
+%https://spectracom.com/resources/essential-education/gps-clock-synchronization
+GPS satellites contain multiple atomic clocks.
+These clocks are monitored and controlled to keep them synchronized with the UTC standard.
+This provides us with an accurate clock to synchronize towards.\cite{gpsclock}
+After testing this, by developing an application to retrieve time from GPS, it was clear to us that the GPS hardware in our phones are limited to 1Hz updates and do not provide millisecond precision, which makes us unable to use GPS as a solution.
+
+\subsubsection{Audio}\mnote{Can go into more detail about how we would solve the difficulties in this, but i do not think it is worth}
+Another possible method is to use the audio to synchronize.
+This would have every device attempt to synchronize based on their perception.
+This method introduces several issues to be addressed, first of all there would be a ramp up time for the devices to figure out synchronization.
+Furthermore adding more devices to the network introduces more complexity as each introduces a new audio source, and with each device figuring out their own synchronization it is unlikely that the devices are synchronization towards the same source.
+While this may not have an effect on humans given only a few milliseconds of variance, due to psychoacoustic effects, it complicates the synchronization of the devices and our possible manipulation.
+
+\subsubsection{Light Cues}
+Alternatively we can use light rather than audio to synchronize.
+This would also allow devices to respond as to achieve synchronization better.
+The idea would be that a device uses the flashlight, once observed by another device it flashes back.
+Using Einstein Synchronization we can use the respective times collected from this, to synchronize the two devices \cite{einstein}.
+This has the significant disadvantage that the facing of the phones becomes important for synchronization, as the phones must have their flashes and cameras facing the synchronization target.

```

Comment:
Et stort problem er også at kameraet er langsomt, de bedste (jeg kender) er iPhones 240 Hz video, og der mangler du stadig noget post processing for at detecte hvornår flashet så skete. Og hvad med vedvarende sync for at undgå drift.